hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/CloudObjectsMetaSelector.java

```diff
@@ -0,0 +1,276 @@
+
+        List<Map> result = new ArrayList<>();
+
+        ReceiveMessageRequest receiveMessageRequest = new ReceiveMessageRequest()
+                .withQueueUrl(this.queueUrl)
+                .withWaitTimeSeconds(this.longPollWait)
+                .withVisibilityTimeout(this.visibilityTimeout);
+        receiveMessageRequest.setMaxNumberOfMessages(this.maxMessagesEachRequest);
+
+        long fetchedMessages = 0;
+
+        // Get count for available messages
+        Map<String, String> queueAttributesResult = getSqsQueueAttributes(sqs);
+        long ApproxMessagesAvailable = Long.parseLong(queueAttributesResult.get("ApproximateNumberOfMessages"));
+        log.info("Approx. " + ApproxMessagesAvailable + "messages available in queue.");
+
+        for (int i = 0; i < (int) Math.ceil((double) ApproxMessagesAvailable / this.maxMessagesEachRequest) + 1; ++i) {
+            List<Message> messages = sqs.receiveMessage(receiveMessageRequest).getMessages();
+            log.debug("Messages size: " + messages.size());
+
+            List<Message> ineligibleMessages = new ArrayList<>();
+
+            for (Message message : messages) {
+                log.debug("message id: " + message.getMessageId());
+                boolean isMessageDelete = Boolean.TRUE;
+
+                JSONObject messageBody = new JSONObject(message.getBody());
+                Map<String, Object> messageMap = new HashMap<>();
+                if (messageBody.has("Message")) {
+                    ObjectMapper mapper = new ObjectMapper();
+                    messageMap = mapper.readValue(messageBody.getString("Message"), Map.class);
+//                    messageBody = new JSONObject(StringEscapeUtils.unescapeJava(
+//                            messageBody.getString("Message")));
+                }
+
+                if (messageBody.has("Records")) {
+                    List<Map<String, Object>> records = (List<Map<String, Object>>) messageMap.get("Records");
+                    for (Map<String, Object> record : records) {
+                        String eventName = (String ) record.get("eventName");
+
+                        // filter only allowed s3 event types
+                        if (ALLOWED_S3_EVENT_PREFIX.stream()
+                                .anyMatch(eventName::startsWith)) {
+
+                            result.add(record);
+                            isMessageDelete = Boolean.FALSE;
+
+                        } else {
+                            log.info("This S3 event " + eventName + " is not allowed, so ignoring it.");
+                        }
+                    }
+                } else {
+                    log.info("Message is not expected format or it's s3:TestEvent");
+                }
+                if (isMessageDelete) {
+                    ineligibleMessages.add(message);
+                }
+            }
+            if (!ineligibleMessages.isEmpty()) {
```

Comment:
Let me know if my understanding is incorrect.
Here, we are trying to only delete ineligible messages (the ones that are not with ALLOWED_S3_EVENT_PREFIX). The eligible messages will get deleted later after the DeltaSync#writeToSink has completed. Is that correct?